Fuse/Classes/Fuse.swift

```diff
@@ -176,7 +176,7 @@ public class Fuse {
             
             // Initialize the bit array
             var bitArr = [Int](repeating: 0, count: finish + 2)
-            bitArr[finish + 1] = (1 << i) - 1
+            bitArr[finish + 1] = (1 << i) &- 1
```

Comment:
Was this intended?

Reply:
This is the fix to stop Swift from trapping from the integer overflow that occurs when the text input is longer than the `Int` type can hold (64 characters on 64 bit devices).

I have added a test `testItHandlesLongInputs()` which failed before making this change.